src/v/redpanda/application.h

```diff
@@ -172,7 +172,6 @@ class application {
     ss::sharded<admin_server> _admin;
     ss::sharded<net::conn_quota> _kafka_conn_quotas;
     ss::sharded<net::server> _kafka_server;
-    ss::sharded<kafka::client::client> _proxy_client;
```

Comment:
This commit may not be necessary in this PR because it's overflowing into other routes that do **not yet** support basic auth. The trade off though is that we'll have a few commits with a kafka client cache and the proxy client in memory until the final PR in this series is merged.

Note that the unit tests and ducktape tests are currently failing because of this commit.

What do people think?